bin/api.wsgi

```diff
@@ -55,75 +38,64 @@ args.site_name = ' '.join(args.site_name).strip('"\'')
     centralclient_enabled = False
     log.warning('central_uri not configured -> SciTran Central functionality disabled')
-if not app.config['drone_secret']:
+if not api.app.config['drone_secret']:
     log.warning('drone_secret not configured -> Drone functionality disabled')
-if not os.path.exists(app.config['data_path']):
-    os.makedirs(app.config['data_path'])
-if not os.path.exists(app.config['quarantine_path']):
-    os.makedirs(app.config['quarantine_path'])
-if not os.path.exists(app.config['upload_path']):
-    os.makedirs(app.config['upload_path'])
-
-for x in range(10):
-    try:
-        app.db = pymongo.MongoClient(args.db_uri).get_default_database()
-    except:
-        time.sleep(6)
-    else:
-        break
-else:
-    raise Exception('Could not connect to MongoDB')
+if not os.path.exists(api.app.config['data_path']):
+    os.makedirs(api.app.config['data_path'])
+if not os.path.exists(api.app.config['quarantine_path']):
+    os.makedirs(api.app.config['quarantine_path'])
+if not os.path.exists(api.app.config['upload_path']):
+    os.makedirs(api.app.config['upload_path'])
 
-# TODO jobs indexes
-# TODO review all indexes
-app.db.projects.create_index([('gid', 1), ('name', 1)])
-app.db.sessions.create_index('project')
-app.db.sessions.create_index('uid')
-app.db.acquisitions.create_index('session')
-app.db.acquisitions.create_index('uid')
-app.db.acquisitions.create_index('collections')
-app.db.authtokens.create_index('timestamp', expireAfterSeconds=600)
-app.db.uploads.create_index('timestamp', expireAfterSeconds=60)
-app.db.downloads.create_index('timestamp', expireAfterSeconds=60)
 
-now = datetime.datetime.utcnow()
-app.db.groups.update_one({'_id': 'unknown'}, {'$setOnInsert': { 'created': now, 'modified': now, 'name': 'Unknown', 'roles': []}}, upsert=True)
-app.db.sites.replace_one({'_id': args.site_id}, {'name': args.site_name, 'api_uri': args.api_uri}, upsert=True)
+api.app.db = mongo.db
 
 
 if __name__ == '__main__':
-    app.debug = True # send stack trace for uncaught exceptions to client
-    paste.httpserver.serve(app, host=args.host, port=args.port, ssl_pem=args.ssl_cert)
+    api.app.debug = True # send stack trace for uncaught exceptions to client
+    paste.httpserver.serve(api.app, host=args.host, port=args.port, ssl_pem=args.ssl_cert)
 else:
-    application = app # needed for uwsgi
+    application = api.app # needed for uwsgi
 
     import uwsgidecorators
 
-    @uwsgidecorators.cron(0, -1, -1, -1, -1)  # top of every hour
-    def upload_storage_cleaning(signum):
-        upload_path = application.config['upload_path']
-        for f in os.listdir(upload_path):
-            fp = os.path.join(upload_path, f)
-            timestamp = datetime.datetime.utcfromtimestamp(int(os.stat(fp).st_mtime))
-            if timestamp < (datetime.datetime.utcnow() - datetime.timedelta(hours=1)):
-                log.debug('upload %s was last modified %s' % (fp, str(timestamp)))
-                os.remove(fp)
-
```

Comment:
the new API won't support sorted upload (you may see that the method isn't in core.py anymore).
This method was cleaning the files created only by that method, as uploads on the */files route automatically removes the temporary files afterwards.